activation/nipst.go

```diff
@@ -204,8 +205,15 @@ func (nb *NIPSTBuilder) BuildNIPST(challenge *types.Hash32) (*types.NIPST, error
 
 	// Phase 1: receive proofs from PoET service
 	if nb.state.PoetProofRef == nil {
-		proofRefChan := nb.poetDb.SubscribeToProofRef(nb.state.PoetServiceId, nb.state.PoetRound.Id)
-		poetProofRef := <-proofRefChan // TODO(noamnelke): handle timeout
+		var poetProofRef []byte
+		select {
+		case poetProofRef = <-nb.poetDb.SubscribeToProofRef(nb.state.PoetServiceId, nb.state.PoetRound.Id):
+		case <-timeout:
+			nb.poetDb.UnsubscribeFromProofRef(nb.state.PoetServiceId, nb.state.PoetRound.Id)
+			return nil, fmt.Errorf("timeout waiting for poet proof, atx target epoch ended")
+		case <-stop:
+			return nil, &StopRequestedError{}
```

Comment:
shouldnt unsubscribe here as well?

Reply:
Not sure what our policy here is - `stop` should happen only when the node shuts down, shouldn't it? What's the point in unsubscribing then? Wdyt?

Reply:
stop could happen if we call stop mining as well no?